### PR TITLE
Improving deployment e2e failure error messages

### PR DESCRIPTION
**Pull request by nikhiljindal**
Ref #21753

Adding more info to failure error messages to help in debugging.

cc @janetkuo @bgrant0607

### CODE REVIEW

File: pkg/util/deployment/deployment.go [hunk 1]
```diff
@@ -206,15 +206,15 @@ func addHashKeyToRSAndPods(deployment *extensions.Deployment, c clientset.Interf
 	// 2. Update all pods managed by the rs to have the new hash label, so they will be correctly adopted.
 	selector, err := unversioned.LabelSelectorAsSelector(updatedRS.Spec.Selector)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error in converting selector to label selector: %s", err)
```
Comment (bgrant0607) [inline pkg/util/deployment/deployment.go]: As long as you're doing this, could you please print identifying information in the message, like updatedRS.Name?

Same for the other log messages, to facilitate searching/filtering for relevant log lines.

**Comment by bgrant0607** [comment]
cc @kargakis

File: pkg/util/deployment/deployment.go (hunk 1 above)
Comment (nikhiljindal) [inline pkg/util/deployment/deployment.go]: Done

**Comment by nikhiljindal** [comment]
Updated code as per comments.
PTAL

File: pkg/util/deployment/deployment.go [hunk 2]
```diff
@@ -303,6 +303,9 @@ func updatePodWithRetries(podClient unversionedcore.PodInterface, pod *api.Pod,
 		}
 		return false, nil
 	})
+	if err != nil && err == wait.ErrWaitTimeout {
```
Comment (janetkuo) [inline pkg/util/deployment/deployment.go]: you don't need `err != nil`

File: pkg/util/deployment/deployment.go [hunk 3]
```diff
@@ -206,15 +206,15 @@ func addHashKeyToRSAndPods(deployment *extensions.Deployment, c clientset.Interf
...
-		return nil, err
+		return nil, fmt.Errorf("error in converting selector to label selector for replica set %s: %s", updatedRS.Name, err)
 	}
 	options := api.ListOptions{LabelSelector: selector}
 	podList, err := getPodList(namespace, options)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error in getting pod list for namespace %s and list options %+v: %s", namespace, options, err)
 	}
 	if err = labelPodsWithHash(podList, c, namespace, hash); err != nil {
```
Comment (janetkuo) [inline pkg/util/deployment/deployment.go]: Improve the returned error from `labelPodsWithHash` as well?

**Comment by janetkuo** [comment]
Feel free to apply LGTM once above two comments are addressed.

File: pkg/util/deployment/deployment.go (hunk 3 above)
Comment (nikhiljindal) [inline pkg/util/deployment/deployment.go]: Done

File: pkg/util/deployment/deployment.go (hunk 2 above)
Comment (nikhiljindal) [inline pkg/util/deployment/deployment.go]: Done

**Comment by nikhiljindal** [comment]
Thanks, updated code as per comments.
Adding the LGTM label

**Comment by bgrant0607** [comment]
Merging to help debug flakes.

**Pull request merged**